src/plugins/discover/public/application/context/context_app.tsx

```diff
@@ -176,12 +175,11 @@ export const ContextApp = ({ dataView, anchorId, referrer }: ContextAppProps) =>
   );
 
   const interceptedWarnings = useMemo(
-    () =>
-      removeInterceptedWarningDuplicates([
-        ...(fetchedState.predecessorsInterceptedWarnings || []),
-        ...(fetchedState.anchorInterceptedWarnings || []),
-        ...(fetchedState.successorsInterceptedWarnings || []),
-      ]),
+    () => [
+      ...(fetchedState.predecessorsInterceptedWarnings || []),
+      ...(fetchedState.anchorInterceptedWarnings || []),
+      ...(fetchedState.successorsInterceptedWarnings || []),
+    ],
```

Comment:
This is a difficult question to answer.

On one hand, each warning is for a different request. Displaying each warning is the only way to provide users with the ability to explore each request/response. If you consolidated into a single warning, then users would only see the shard failures/request/response for one of the request and not all 3.

On the other hand, making 3 requests is an implementation detail that the user does not need to know about (or care about). If you consolidate into a single warning - what request/response would "Show details" link to?

If I had to pick an option, I would go with displaying a warning per request. This exposes all of the requests/responses to users so they can fully debug the situation. Users will not be overwhelmed with warnings because the number of warnings is caped to the number of requests - which is only  3.

Reply:
All good points! Unfortunately, it can be more than 3 displayed warnings as successors and predecessors parts can fetch data for multiple intervals https://github.com/nreese/kibana/blob/ee013d67994079704b9a4ebbd36bd54f0d64b386/src/plugins/discover/public/application/context/services/context.ts#L77

@davismcphee Would it be okay to show all callouts for such cases? Might be overwhelming for users but at least the UI section for callouts is scrollable. If we decide to hide similar warnings, I think showing request/response for the first of them could be enough information for users to start the debugging.

Reply:
My experience with the interval logic for surrounding docs is limited, but I think it's generally better to favour more access to debug info for users than reduced UI clutter if it means missing info. We probably should do something about this at some point, but I'm happy to take a phased approach to it and consider this the first pass. So unless either of you have strong objections, I'm ok with including banners for each failure for now.

If there's a way to differentiate each banner in the list (e.g. `This data might be incomplete or wrong (successor request 1 of X)`), that might be a good option for now too, but I'm also ok will leaving it how it is for this PR.